File: api/src/main/java/me/iska/jclient/user/Permission.java
package me.iska.jclient.user;

/**
 * A permission for a user, contains the name of the permission and the level.
 */
public class Permission {

    private final String name;
    private final int level;

    public Permission(String name, int level) {
        this.name = name;
        this.level = level;
    }

    @Override
    public String toString() {
        return String.format("Permission(name=%s, level=%d)", name, level);
    }

    public String getName() {
        return name;
    }

    public int getLevel() {
        return level;
    }
}

File: viewer/pyclient/impl/permission.py
#!/usr/bin/env python3


class Permission:
    """
    A permission for a user, contains the name and level of the permission.
    """

    @property
    def name(self) -> str:
        return self._name

    @property
    def level(self) -> int:
        return self._level

    def __init__(self, name: str, level: int) -> None:
        self._name = name
        self._level = level

    def __repr__(self) -> str:
        return "Permission(name=%s, level=%i)" % (self._name, self._level)
